server/api/config_test.go

```diff
@@ -278,28 +278,62 @@ func (s *testConfigSuite) TestConfigDefault(c *C) {
+	c.Assert(svr.GetPersistOptions().GetMaxMergeRegionKeys(), checker, uint64(999))
+	c.Assert(svr.GetPersistOptions().GetSchedulerMaxWaitingOperator(), checker, uint64(999))
+	c.Assert(svr.GetPersistOptions().GetLeaderScheduleLimit(), checker, uint64(999))
+	c.Assert(svr.GetPersistOptions().GetRegionScheduleLimit(), checker, uint64(999))
+	c.Assert(svr.GetPersistOptions().GetHotRegionScheduleLimit(), checker, uint64(999))
+	c.Assert(svr.GetPersistOptions().GetReplicaScheduleLimit(), checker, uint64(999))
+	c.Assert(svr.GetPersistOptions().GetMergeScheduleLimit(), checker, uint64(999))
+}
+
 func (s *testConfigSuite) TestConfigTTL(c *C) {
-	addr := fmt.Sprintf("%s/config?ttlSecond=3", s.urlPrefix)
-	r := map[string]interface{}{
-		"schedule.max-snapshot-count":             999,
-		"schedule.enable-location-replacement":    false,
-		"schedule.max-merge-region-size":          999,
-		"schedule.max-merge-region-keys":          999,
-		"schedule.scheduler-max-waiting-operator": 999,
-	}
-	postData, err := json.Marshal(r)
+	addr := fmt.Sprintf("%s/config?ttlSecond=2", s.urlPrefix)
+	postData, err := json.Marshal(TTLConfig)
 	c.Assert(err, IsNil)
 	err = postJSON(testDialClient, addr, postData)
 	c.Assert(err, IsNil)
-	c.Assert(s.svr.GetPersistOptions().GetMaxSnapshotCount(), Equals, uint64(999))
-	c.Assert(s.svr.GetPersistOptions().IsLocationReplacementEnabled(), Equals, false)
-	c.Assert(s.svr.GetPersistOptions().GetMaxMergeRegionSize(), Equals, uint64(999))
-	c.Assert(s.svr.GetPersistOptions().GetMaxMergeRegionKeys(), Equals, uint64(999))
-	c.Assert(s.svr.GetPersistOptions().GetSchedulerMaxWaitingOperator(), Equals, uint64(999))
-	time.Sleep(5 * time.Second)
-	c.Assert(s.svr.GetPersistOptions().GetMaxSnapshotCount(), Not(Equals), uint64(999))
-	c.Assert(s.svr.GetPersistOptions().IsLocationReplacementEnabled(), Equals, true)
-	c.Assert(s.svr.GetPersistOptions().GetMaxMergeRegionSize(), Not(Equals), uint64(999))
-	c.Assert(s.svr.GetPersistOptions().GetMaxMergeRegionKeys(), Not(Equals), uint64(999))
-	c.Assert(s.svr.GetPersistOptions().GetSchedulerMaxWaitingOperator(), Not(Equals), uint64(999))
+	assertTTLConfig(c, s.svr, Equals)
+	time.Sleep(3 * time.Second)
+	assertTTLConfig(c, s.svr, Not(Equals))
+}
+
+func (s *testConfigSuite) TestConfigTTLAfterTransferLeader(c *C) {
+	_, svrs, cleanup := mustNewCluster(c, 3)
+	defer cleanup()
+	leader := mustWaitLeader(c, svrs)
+	addr := fmt.Sprintf("%s/pd/api/v1/config?ttlSecond=10", leader.GetAddr())
+	postData, err := json.Marshal(TTLConfig)
+	c.Assert(err, IsNil)
+	err = postJSON(testDialClient, addr, postData)
+	c.Assert(err, IsNil)
+	time.Sleep(3 * time.Second)
+	leader.Close()
+	// remove the old leader from svrs, or next mustWaitLeader will fail
+	for i, svr := range svrs {
+		if svr == leader {
+			svrs = append(svrs[:i], svrs[i+1:]...)
+			break
+		}
+	}
+	time.Sleep(3 * time.Second)
+	leader = mustWaitLeader(c, svrs)
+	assertTTLConfig(c, leader, Equals)
```

Comment:
move to `tests/client/client_test.go`

Reply:
This is hard because we have to depend a lot of util functions, eg. `mustWaitLeader` in `api` package.

As it do tests ConfigTTL, I think it's ok to leave it here.